projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -231,6 +231,12 @@ export class IgxFilteringService implements OnDestroy {
         if (field) {
             const expressions = this.getExpressions(field);
             expressions.length = 0;
+        } else {
+            const fields = this.grid.columns.map(c => c.field);
+            fields.forEach(f => {
+                const expressions = this.getExpressions(f);
+                expressions.length = 0;
+            });
```

Comment:
You can "optimize" this:
```suggestion
            this.grid.columns.forEach(c => {
                const expressions = this.getExpressions(c.field);
                expressions.length = 0;
            });
```